### ISSUE TITLE
Passenger 4 Issue with capistrano-passenger 0.0.4

### ISSUE THREAD

**Issue by exterm** [capistrano/passenger#7]
We currently deploy with capistrano-passenger 0.0.2 . When we update capistrano-passenger to 0.0.4, we get the following error:

```
DEBUG [cd2a345d] Running /usr/bin/env passenger -v on 5.22.150.180
DEBUG [cd2a345d] Command: /usr/bin/env passenger -v
DEBUG [cd2a345d]     /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to_specs'
DEBUG [cd2a345d]     : 
DEBUG [cd2a345d]     Could not find 'passenger' (>= 0) among 15 total gem(s)
DEBUG [cd2a345d]      (
DEBUG [cd2a345d]     Gem::LoadError
DEBUG [cd2a345d]     )
DEBUG [cd2a345d]     
DEBUG [cd2a345d]         
DEBUG [cd2a345d]     from /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:309:in `to_spec'
DEBUG [cd2a345d]     
DEBUG [cd2a345d]         
DEBUG [cd2a345d]     from /usr/local/lib/ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:53:in `gem'
DEBUG [cd2a345d]     
DEBUG [cd2a345d]         
DEBUG [cd2a345d]     from /usr/local/bin/passenger:22:in `<main>'
DEBUG [cd2a345d]     
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing on host 5.22.150.180: passenger exit status: 1
passenger stdout: /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to_specs': Could not find 'passenger' (>= 0) among 15 total gem(s) (Gem::LoadError)
    from /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:309:in `to_spec'
    from /usr/local/lib/ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:53:in `gem'
    from /usr/local/bin/passenger:22:in `<main>'
passenger stderr: Nothing written

SSHKit::Command::Failed: passenger exit status: 1
passenger stdout: /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to_specs': Could not find 'passenger' (>= 0) among 15 total gem(s) (Gem::LoadError)
    from /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:309:in `to_spec'
    from /usr/local/lib/ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:53:in `gem'
    from /usr/local/bin/passenger:22:in `<main>'
passenger stderr: Nothing written

Tasks: TOP => passenger:restart
(See full trace by running task with --trace)
The deploy has failed with an error: Exception while executing on host 5.22.150.180: passenger exit status: 1
passenger stdout: /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to_specs': Could not find 'passenger' (>= 0) among 15 total gem(s) (Gem::LoadError)
    from /usr/local/lib/ruby/2.1.0/rubygems/dependency.rb:309:in `to_spec'
    from /usr/local/lib/ruby/2.1.0/rubygems/core_ext/kernel_gem.rb:53:in `gem'
    from /usr/local/bin/passenger:22:in `<main>'
passenger stderr: Nothing written
```

Downgrading to 0.0.2 fixes the problem.

We have Passenger 4 installed on our servers.

**Comment by betesh** [comment]
The most likely issue here is that `/usr/bin/env passenger -v` needs some prefix on the command map--something similar to what [this task](https://github.com/capistrano/passenger/blob/master/lib/capistrano/tasks/passenger.cap#L22) handles for RVM users.

Are you setting `passenger_environment_variables` as described in the [README](https://github.com/capistrano/passenger#restarting-passenger-5-and-above-applications)?  (I realize it's under the Passenger 5 section, but `passenger -v` is run to determine the version, so we should probably move that to another section of the README.)

When SSH'ed into the server, what do you need to do to run `passenger -v`?  What's in your $PATH in an SSH session that's not there for capistrano?  If we can answer that question, and it's a very common situation, we can add another hook just like the one for RVM so that no custom configuration is necessary.

**Comment by exterm** [comment]
Hey, thanks for your answer!

We are not setting `passenger_environment_variables`. Should we? Also, we are using `rbenv`, maybe there needs to be something similar to the `rvm` integration you mentioned, but for `rbenv`?

**Comment by betesh** [comment]
I suspect you are on the right track about rbenv.  Unfortunately, I am not familiar with rbenv, but I took a look at the source code for capistrano-rbenv and put together a solution that I suspect will work.  I pushed it to master and now I need someone using rbenv to test it and let me know if it works.

**Comment by exterm** [comment]
Just tested it, works for me!

```
DEBUG [fc829158] Running /usr/bin/env passenger -v on 144.76.112.16
DEBUG [fc829158] Command: /usr/bin/env passenger -v
DEBUG [fc829158]    Phusion Passenger version 4.0.57
DEBUG [fc829158]    
DEBUG [fc829158]    "Phusion Passenger" is a trademark of Hongli Lai & Ninh Bui.
DEBUG [fc829158] Finished in 0.400 seconds with exit status 0 (successful).
```

**Comment by betesh** [comment]
In that case, for rbenv users, I believe this is fixed by bc86590d73

I will release a new gem once we've debugged what's going on in #2, but until then, feel free to use master.

**Comment by exterm** [comment]
Thanks for the fast fix!
